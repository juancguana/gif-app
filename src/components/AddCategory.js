import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (inputValue.trim().length >= 2) {
      setCategories((categories) => [inputValue, ...categories]);
      setInputValue('');
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input type='text' value={inputValue} onChange={handleInputChange} />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export { AddCategory };
