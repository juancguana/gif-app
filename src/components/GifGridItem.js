import React from 'react';
import PropTypes from 'prop-types';

const GifGridItem = ({ title, url }) => {
  return (
    <figure className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export { GifGridItem };
