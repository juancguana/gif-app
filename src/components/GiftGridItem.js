import React from 'react';
import PropTypes from 'prop-types';

const GiftGridItem = ({ title, url }) => {
  return (
    <figure className='card animate__animated animate__fadeIn'>
      <img src={url} alt={title} />
      <figcaption>{title}</figcaption>
    </figure>
  );
};

GiftGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GiftGridItem;
