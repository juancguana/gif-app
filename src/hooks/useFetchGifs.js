import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((imgs) => {
      setState({
        data: imgs,
        loading: false,
      });
    }).catch(error => {
      console.error(error.message);
    });
  }, [category]);

  return state;
};
