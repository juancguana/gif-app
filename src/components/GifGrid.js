import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>
        {category.toUpperCase()}
      </h3>
      {loading && (
        <figure className='animate__animated animate__flash loading'>
          <img src='/logo.png' height={100} alt='Logo' />
        </figure>
      )}
      <main className='card-grid'>
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </main>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export { GifGrid };
