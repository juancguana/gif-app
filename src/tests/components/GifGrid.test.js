import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Test in <GifGrid />', () => {
  const category = 'spiderman';

  test('should show the component correctly', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when load images useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'https://localhost/a/1',
        title: 'Title',
      },
      {
        id: 'DDAS',
        url: 'https://localhost/a/1',
        title: 'Title',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
