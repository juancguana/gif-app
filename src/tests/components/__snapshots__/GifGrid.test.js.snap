// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid /> should show the component correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    SPIDERMAN
  </h3>
  <p
    className="animate__animated animate__flash loading"
  >
    Loading...
  </p>
  <main
    className="card-grid"
  />
</Fragment>
`;
